test/contrib/test_module.py

```diff
@@ -207,6 +207,65 @@ def model(data, labels):
     )
 
 
+def test_random_module_mcmc_callable(backend, init):
+
+    if backend == "flax":
+        import flax
+
+        linear_module = flax.linen.Dense(features=1)
+        bias_name = "bias"
+        weight_name = "kernel"
+        random_module = random_flax_module
+        kwargs_name = "inputs"
+    elif backend == "haiku":
+        import haiku as hk
+
+        linear_module = hk.transform(lambda x: hk.Linear(1)(x))
+        bias_name = "linear.b"
+        weight_name = "linear.w"
+        random_module = random_haiku_module
+        kwargs_name = "x"
+
+    N, dim = 3000, 3
+    num_warmup, num_samples = (1000, 1000)
+    data = random.normal(random.PRNGKey(0), (N, dim))
+    true_coefs = np.arange(1.0, dim + 1.0)
+    logits = np.sum(true_coefs * data, axis=-1)
+    labels = dist.Bernoulli(logits=logits).sample(random.PRNGKey(1))
+
+    if init == "shape":
+        kwargs = {"input_shape": (3,)}
+    elif init == "kwargs":
+        kwargs = {kwargs_name: data}
+
+    def model(data, labels):
+        nn = random_module(
+            "nn",
+            linear_module,
+            prior=(lambda name, shape: dist.Cauchy() if name == "bias_name" else dist.Normal()),
```

Comment:
@Carlosbogo Are you still working on this PR? I guessed you got stuck with making this test pass? FYI, in pytest, you can parameterize so that you can run the same code with different parameters, e.g. those parameters are `backend` and `init` arguments of this test. To simplify, here are my suggestions:
+ Add a new argument to `test_random_module_mcmc`:
```python
@pytest.mark.parametrize("backend", ["flax", "haiku"])
@pytest.mark.parametrize("init", ["shape", "kwargs"])
@pytest.mark.parametrize("callable_prior", [True, False])
def test_random_module_mcmc(backend, init, callable_prior):
```
+ Set prior depending on `callable_prior`:
```python
if callable_prior:
    prior = (lambda name, shape: dist.Cauchy() if name == bias_name else dist.Normal())
else:
    prior = {bias_name: dist.Cauchy(), weight_name: dist.Normal()}
```
+ Then you can define the model as follows
```python
    def model(data, labels):
        nn = random_module("nn", linear_module, prior=prior, **kwargs)
        logits = nn(data).squeeze(-1)
        numpyro.sample("y", dist.Bernoulli(logits=logits), obs=labels)
```

Reply:
Sorry for the delay, after trying to do it for a while my exams started and I didn't have time to work on it.

I just added the changes you mentioned. However, I am still getting used to how the local Git and Github relate to each other (I'm also trying to use Github desktop to see if it's worth it, but there are still a lot of things I'm not sure about), so I don't know whether the commit I just made to my local clone of the repository is enough or if I should create another PR (or if I can do something to push the changes into this one).

Thank you so much for the tips, they are really helpful and sorry again for the disturbance.

Reply:
No worries! This contribution is great. I love to use this new `prior` pattern. :)